pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -140,18 +150,19 @@ public void initialize(URL location, ResourceBundle resources) {
 
         xpathExpressionArea.addEventHandler(KeyEvent.KEY_PRESSED, t -> {
             if ((t.isControlDown() && t.getCode().isWhitespaceKey()) || t.getCode() == KeyCode.BACK_SPACE) {
-                autoCompletePopup.show(xpathExpressionArea, 500, 500);
+                autoCompletePopup.show(xpathExpressionArea, xpathExpressionArea.getCharacterBoundsOnScreen(positionFrom, positionTo)
```

Comment:
Use a local variable to not duplicate the call to getCharacterBoundsOnScreen. And as I already said, move the call to `show` inside the autoComplete method and merge the stream of ctrl+space events into the main processing stream.